modules/dxp/apps/saml/saml-web/src/test/java/com/liferay/saml/web/internal/struts/AssertionConsumerServiceActionTest.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.saml.web.internal.struts;
+
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+import com.liferay.saml.runtime.configuration.SamlProviderConfigurationHelper;
+import com.liferay.saml.runtime.servlet.profile.WebSsoProfile;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+/**
+ * @author Manuele Castro
+ */
+public class AssertionConsumerServiceActionTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@BeforeClass
+	public static void setUpClass() {
+		_setUpSamlProviderConfigurationHelper();
+		_setUpWebSsoProfile();
+
+		_setUpAssertionConsumerServiceAction();
+	}
+
+	@Test
+	public void testExecute() throws Exception {
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest(
+				"GET", "http://localhost:8080/c/portal/saml/acs");
+
+		Assert.assertEquals(
+			"/common/referer_js.jsp",
+			_assertionConsumerServiceAction.execute(
+				mockHttpServletRequest, new MockHttpServletResponse()));
+
+		Mockito.verify(
+			_webSsoProfile, Mockito.times(0)
+		).processResponse(
+			Mockito.any(), Mockito.any()
+		);
+	}
+
+	private static void _setUpAssertionConsumerServiceAction() {
+		_assertionConsumerServiceAction = new AssertionConsumerServiceAction();
+
+		ReflectionTestUtil.setFieldValue(
+			_assertionConsumerServiceAction, "_samlProviderConfigurationHelper",
+			_samlProviderConfigurationHelper);
+
+		ReflectionTestUtil.setFieldValue(
+			_assertionConsumerServiceAction, "_webSsoProfile", _webSsoProfile);
+	}
+
+	private static void _setUpSamlProviderConfigurationHelper() {
+		_samlProviderConfigurationHelper = Mockito.mock(
+			SamlProviderConfigurationHelper.class);
+
+		Mockito.when(
+			_samlProviderConfigurationHelper.isRoleSp()
+		).thenReturn(
+			true
+		);
+
+		Mockito.when(
+			_samlProviderConfigurationHelper.isEnabled()
+		).thenReturn(
+			true
+		);
+	}
+
+	private static void _setUpWebSsoProfile() {
+		_webSsoProfile = Mockito.mock(WebSsoProfile.class);
+	}
+
+	private static AssertionConsumerServiceAction
```

Comment:
Hey @manuelecastro, these fields don't need to be static. Using a `setUp` method with a `@Before` is enough here.

Thanks

Reply:
Also, make all other methods non-static